### ISSUE TITLE
Expose public API method to add canned files

### ISSUE THREAD

**Issue by rnicholus** [FineUploader/fine-uploader#1191]
Currently, canned file data must come from a server.  If the initial file list feature is enabled, Fine Uploader sends a GET request on init (and on `reset`), expecting JSON data to represent files uploaded successfully in a previous session.  It might be useful to allow successfully upload metadata to be persisted to and read from `localStorage` as an alternative.  Obviously, this is not possible in browsers that don't support `localStorage` such as IE7.

Another option may be to provide an API method where integrators can passed canned file data at any time.  This allows the data to be stored anywhere.

**Comment by rnicholus** [comment]
@ludofleury & @alphestdev Can one or both of you commit to verifying that that API method I will add in this issue addresses your needs?

**Comment by alphestdev** [comment]
@rnicholus I'll gladly implement and verify this as soon as it's released.

**Comment by rnicholus** [comment]
...this would be pre-release testing. I'll make sure you get access to the pre-release version and will wait for your feedback before release.

**Comment by rnicholus** [comment]
[I've pushed a pre-release (not on npm) - 5.6.0-1](https://s3.amazonaws.com/releases.fineuploader.com/develop/5.6.0-1/fine-uploader-5.6.0-1.zip) which contains a [new API method - `addInitialFiles`](http://docs.fineuploader.com/branch/develop/api/methods.html#addInitialFiles). Please let me know if this is working as expected. Upon verification, I'll formally release this as Fine Uploader 5.6.0.

**Comment by alphestdev** [comment]
Hi Ray. The method you provided in v5.6.0.1 works well, but does not meet my needs because it doesn't return the id's of added files, which we use to maintain our list of added files to prevent duplicates (like you suggested somewhere on stackoverflow) and other needs.

Take for example our current implementation:
```
var addedFiles = []; // Keep track of all added files
var uploader = ...;
$$('.js-file-upload-initial-file[data-inputname="'+ inputName +'"]').each(function(hiddenField){
  var id = uploader._addCannedFile({
    uuid: hiddenField.get('data-uuid'),
    name: hiddenField.get('data-filename'),
    size: hiddenField.get('data-size'),
    thumbnailUrl: hiddenField.get('data-thumbnail-url')
  });
  addedFiles[id] = [hiddenField.get('data-filename'), hiddenField.get('data-size'), hiddenField];
});
uploader._templating.addCacheToDom();
```

This cannot be programmed using only `addInitialFiles(files)`, because this function does not return id's even if the canned files are added one by one.

**Comment by rnicholus** [comment]
You can get a list of all files tracked by the uploader (as well as each file's name, size, uuid, id, etc) using the `getUploads` API method.

**Comment by alphestdev** [comment]
Using `getUploads` method I get the added file records with id's correctly. It's not performance friendly because now I have 2 loops instead of one, but it works.

**Comment by rnicholus** [comment]
Unless you are looping through thousands of files, there is likely no perf difference, but in really not clear why you are loophole by over anything. I'd advise stepping back and taking a close lo K at your code to see how it can be improved. Perhaps there is a much better approach to solve whatever problem you are trying to solve, given the available API. 

Thanks for the update, I will release this new version next week.

**Comment by rnicholus** [comment]
Also if you post a question on stack overflow under the fine-uploader tag, explain what you are trying to do with all of these loops (taking into account use of the API method I just added), and show all of your code, I can help you find a better approach as well.

**Comment by rnicholus** [comment]
This has been released as part of 5.6.0.
